src/ui/profile_preferences.ui

```diff
@@ -568,27 +568,109 @@
        <item row="1" column="0">
         <widget class="QGroupBox" name="groupBox_spellCheck">
          <property name="title">
-          <string>Spell check</string>
+          <string>Spell checking</string>
          </property>
-         <layout class="QGridLayout" name="gridLayout_12">
-          <item row="0" column="0">
-           <widget class="QLabel" name="label_57">
+         <property name="checkable">
+          <bool>true</bool>
+         </property>
+         <property name="checked">
+          <bool>false</bool>
+         </property>
+         <layout class="QVBoxLayout" name="verticalLayout_2">
+          <item>
+           <widget class="QLabel" name="label_chooseSpellCheckDictionary">
             <property name="text">
              <string>Choose spell check dictionary:</string>
             </property>
            </widget>
           </item>
-          <item row="0" column="1">
-           <widget class="QListWidget" name="dictList"/>
+          <item>
+           <widget class="QListWidget" name="dictList">
+            <property name="flow">
+             <enum>QListView::TopToBottom</enum>
+            </property>
+            <property name="resizeMode">
+             <enum>QListView::Adjust</enum>
+            </property>
+            <property name="viewMode">
+             <enum>QListView::IconMode</enum>
+            </property>
+            <property name="sortingEnabled">
+             <bool>true</bool>
+            </property>
+           </widget>
           </item>
-          <item row="1" column="0" colspan="2">
-           <widget class="QCheckBox" name="enableSpellCheck">
-            <property name="text">
-             <string>Enable spell check</string>
+          <item>
+           <widget class="QGroupBox" name="groupBox_autoAddWordsFromServerToDictionary">
+            <property name="title">
+             <string>Automatically add words from server</string>
             </property>
-            <property name="checked">
+            <property name="checkable">
              <bool>true</bool>
             </property>
+            <property name="checked">
+             <bool>false</bool>
```

Comment:
Before this PR, default was spell check enabled. Did we loose this now?

Reply:
That is a bad idea (IMVHO) because not all languages *have* spell checking support - I have come to realise that some of the ideographic scripts (think Far-Eastern!) cannot currently be handled by Hunspell.  As it is, I am just running through the list of languages that my Linux distribution supports to see whether it would be reasonable to grab them to fix #2291 ...

Reply:
It turns out it is only like that if there is not a `Host` instance around - to leave the display blank and disabled - it gets set to the current state of the relevant `Host` when that is used to initialise the preferences....